### ISSUE TITLE
Star Wars Galactic Battlegrounds Saga (356500)

### ISSUE THREAD

**Issue by CentralPython** [ValveSoftware/Proton#1427]
# Compatibility Report
- Name of the game with compatibility issues: Star Wars Galactic Battlegrounds Saga
- Steam AppID of the game: 356500

## System Information
- GPU: GTX 750 Ti
- Driver/LLVM version: nvidia 396.54
- Kernel version: 4.15 
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/FederalRecall/cd98b62cde3ae0ebdb11fbc8909a1559
- Proton version: 3.7-6-Beta

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

[steam-356500.log](https://github.com/ValveSoftware/Proton/files/2380016/steam-356500.log)

## Symptoms 

I clicked on Play on steam it only states Running and then blinks and then dumps back to the desktop with no game window opening and nothing happens.

## Reproduction

Activate the following on Steam Play: 
- Enable Steam Play for All Titles
- Use this tool instead of Game-specific selections from steam 
- Change selection from Compatibility Tool to Proton Beta 
- Download the game and then click Play and let it install necessary files like DirectX etc. then actual crash happens after finishing final setup.

**Comment by chriseilander** [comment]
I remembered that this game has a launcher in which you can select which of the games to launch. This launcher crashes immediately. I think this might be similar to Age of Empires 2 HD where it is also unable to launch the launcher.

In the game folder there is an folder named /game/ which contains the executables. "player.exe" for the launcher, "Battlegrounds.exe" for the original game and "battlegrounds_x1.exe" for the Clone Campaigns.

Renaming one of those 2 to player.exe will result in a hanging black screen. I tried deleting the intro movies as I thought that might be an issue, but it keeps hanging.

**Comment by asoltesz** [comment]
This 2018-APR howto says that the WineCfg "Emulate a virtual desktop" setting is needed to start the game in Wine:

https://appuals.com/play-star-wars-galactic-battlegrounds-wine-linux/

I am unsure how that setting can be replicated in Proton (which is a close Wine fork, I believe), just thought I would mention it, maybe helps solving it under proton.

**Comment by chriseilander** [comment]
Look a bit around and the "Emulate a virtual desktop" can be enabled on Proton using the following terminal command:

`WINEPREFIX=/home/username/.steam/steam/steamapps/compatdata/356500/pfx/ winecfg`

For the resolution, I've kept the defaults.

This still won't boot the game. As mentioned earlier, rename "player.exe" to something else and rename "Battlegrounds.exe" (or "battlegrounds_x1.exe" for the expansion) to "player.exe".

Doing that will boot the game. The menu has some graphical issues and for some reason I couldn't scroll upwards (had to use the arrow keys). 

I played through the first mission of OOM-9's campaign and it seems to work okay.

[edit]
Basically the top part of the emulated desktop is unusable, and it does depend on the selected resolution both in-game and for the emulated desktop to how much is usable.

**Comment by dreamer** [comment]
Have anyone testing earlier had working intro video? With 4.2-3 launcher seems to be "working" (at least partially).

Anyway, decided to test it using 4.2-3 and game closed immediately - log contained an error about "vidc IV50" codec missing. I installed `iv5setup.exe` from [here](https://www.moviecodec.com/solutions/iv31-iv32-iv41-iv50-indeo-3-4-and-5-8211/) (during installation unselect everything not related to 5.0 codec - especially browser plugins) and game the started, but intro was rendered in a tiny window. Then I ran `protontricks 356500 winecfg` and enabled virtual desktop. After that game starts (in window - weird); window needs to be moved to the left side of the screen, otherwise menu options won't be reachable with cursor.

**Comment by beedell-roke** [comment]
I've been unable to launch it, as of yet. I've tried Proton 8 and Experimental.



<blockquote>

```YAML
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
/bin/sh\0-c\0/home/rokejulianlockhart/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=356500 -- /home/rokejulianlockhart/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun -- '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental'/proton waitforexitandrun  '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/STAR WARS - Galactic Battlegrounds Saga/Game/player.exe' xlogo1.avi xintro.avi battlegrounds_x1.exe\0
chdir "/home/rokejulianlockhart/.local/share/Steam/steamapps/common/STAR WARS - Galactic Battlegrounds Saga/Game"
ERROR: ld.so: object '/home/rokejulianlockhart/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/rokejulianlockhart/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
ERROR: ld.so: object '/home/rokejulianlockhart/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/rokejulianlockhart/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/rokejulianlockhart/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
Setting breakpad minidump AppID = 356500
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198832168690 [API loaded no]

(wine:31566): GStreamer-WARNING **: 18:04:54.549: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstadder.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstadder.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.549: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstapp.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstapp.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.549: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstaudioconvert.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstaudioconvert.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.549: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstaudioparsers.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstaudioparsers.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.549: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstaudioresample.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstaudioresample.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.549: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstautodetect.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstautodetect.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.549: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstavi.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstavi.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstcoreelements.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstcoreelements.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstcoretracers.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstcoretracers.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstdebug.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstdebug.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstdeinterlace.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstdeinterlace.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstid3demux.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstid3demux.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstisomp4.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstisomp4.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstlibav.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstlibav.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstmatroska.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstmatroska.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstmpg123.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstmpg123.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstnavigationtest.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstnavigationtest.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstogg.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstogg.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstopengl.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstopengl.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstopus.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstopus.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstpbtypes.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstpbtypes.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstplayback.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstplayback.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstrsdav1d.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstrsdav1d.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstsoup.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstsoup.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgsttheora.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgsttheora.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.550: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgsttypefindfunctions.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgsttypefindfunctions.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.551: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstvideobox.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstvideobox.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.551: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstvideoconvert.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstvideoconvert.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.551: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstvideofilter.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstvideofilter.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.551: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstvideoscale.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstvideoscale.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.551: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstvorbis.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstvorbis.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.551: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstvpx.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstvpx.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.551: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstwavparse.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstwavparse.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.551: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstximagesink.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libgstximagesink.so: wrong ELF class: ELFCLASS64

(wine:31566): GStreamer-WARNING **: 18:04:54.551: Failed to load plugin '/home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libprotonmediaconverter.so': /home/rokejulianlockhart/.local/share/Steam/steamapps/common/Proton - Experimental/files/lib64/gstreamer-1.0/libprotonmediaconverter.so: wrong ELF class: ELFCLASS64
pid 31441 != 31440, skipping destruction (fork without exec?)
Game 356500 created interface STEAMAPPTICKET_INTERFACE_VERSION001 / 
Game 356500 created interface SteamUser017 / User
Game 356500 created interface SteamUtils007 / Utils
Game 356500 method call count for IClientUtils::RecordSteamInterfaceCreation : 3
Game 356500 method call count for IClientUtils::GetAppID : 4
Game 356500 method call count for IClientUser::GetAppOwnershipTicketExtendedData : 1
Game 356500 method call count for IClientUser::GetSteamID : 1
Uploaded AppInterfaceStats to Steam
```

</blockquote>

**Comment by ranplayer** [comment]
So this game won't launch with Proton 8, 9 and 10 unless some workarounds are applied:
1. Backup the file `player.exe` (directory: `STAR WARS - Galactic Battlegrounds Saga/Game`)
 2. Create a symlink to `battlegrounds_x1.exe` named `player.exe` 

The steps above already fix the launch, but some chars and content are misrendered. To fix this we need to translate D3D7 calls to D3D9 with [dxwrapper](https://github.com/elishacloud/dxwrapper)

4. Download [dxwrapper.zip](https://github.com/elishacloud/dxwrapper/releases/tag/v1.3.7700.25) and extract the following files to  **STAR WARS - Galactic Battlegrounds Saga/Game**: `ddraw.dll`, `dxwrapper.dll` and `dxwrapper.ini`
5. Edit `dxwrapper.ini` and set `Dd7to9=1` (optionally enable vsync by setting `EnableVSync=1` in `[d3d9]`) 
6. Edit the games Launch Properties to `WINEDLLOVERRIDES="ddraw=n" %command%`

**Comment by wez** [comment]
I'm using ranplayer's method of making the game launch.

In order to make multiplayer (even just being able to setup the LAN lobby) work, I did some hunting and found this gist https://gist.github.com/nils-ballmann/ce467beaf4d10b8fa3ad54baf1f8933b

The instructions there didn't work for me, but I did transcribe them to the following steps, which reinstall directplay via protontricks:

```
$ sudo flatpak install com.github.Matoking.protontricks
$ cd ~/.local/share/Steam/steamapps/compatdata/356500/pfx/drive_c
$ rm windows/*/{dplaysvr.exe,dplayx.dll,dpnet.dll,dpnhpast.dll,dpnsvr.exe,dpwsockx.dll}
$ flatpak run com.github.Matoking.protontricks 356500 directplay
```

with that done, now the LAN multiplayer game dialog actually launches for me, and I can successfully play a LAN game.

I will note that I had to select "Force the use of a specific Steam Play compatibility tool" and a version of proton for protontricks to run; I selected "Proton Experimental" for that.
